docs/dictionary/function/processor.lcdoc

```diff
@@ -23,17 +23,15 @@ if the processor is "ARM" then ...
 
 Returns (enum):
 
--   x86: Windows systems (any architecture) and Mac OS X and Linux
-    systems with a 32 bit processor.
--   x86_64: Mac and Linux systems running a 64 bit processor
--   i386: iOS Simulator.
--   ARM: iOS and Android Devices and the Android simulator
-
+-   x86: 32 or 64 bit Windows systems and 32 bit macOS or Linux systems.
```

Comment:
@seaniepie The docs were wrong here for Windows. It will return the architecture the engine was compiled for rather than the processor it is running on. This is a bit of an anomaly as we really should be using something like this to get the processor https://msdn.microsoft.com/en-us/library/ms724958(v=vs.85).aspx